catalog/ui/search-ui/search-endpoint/src/main/java/org/codice/ddf/ui/searchui/query/model/QueryStatus.java

```diff
@@ -98,6 +103,18 @@ public boolean isDone() {
         return !(state == State.ACTIVE);
     }
 
+    public boolean hasReasons() {
+        return !reasons.isEmpty();
+    }
+
+    public List<String> getReasons() {
+        return Collections.unmodifiableList(reasons);
+    }
+
+    public void addReason(String reason) {
+        this.reasons.add(reason);
```

Comment:
The only reason I used `this.reasons` is to make it perfectly clear that I'm modifying state. It might be confusing if someone skims over it and sees `reason` as an argument and `reasons` as the list. Just a precaution.